zcash_client_sqlite/src/prepared.rs

```diff
@@ -119,22 +165,26 @@ impl<'a, P> DataConnStmtCache<'a, P> {
                         received_by_account, address,
                         prevout_txid, prevout_idx, script,
                         value_zat, height)
-                    VALUES (
-                        :received_by_account, :address,
+                    SELECT
+                        addresses.account, :address,
                         :prevout_txid, :prevout_idx, :script,
-                        :value_zat, :height)
+                        :value_zat, :height
+                    FROM addresses
+                    WHERE addresses.cached_transparent_receiver_address = :address
                     RETURNING id_utxo"
                 )?,
                 #[cfg(feature = "transparent-inputs")]
                 stmt_update_received_transparent_utxo: wallet_db.conn.prepare(
                     "UPDATE utxos
-                    SET received_by_account = :received_by_account,
+                    SET received_by_account = addresses.account,
                         height = :height,
                         address = :address,
                         script = :script,
                         value_zat = :value_zat
+                    FROM addresses
                     WHERE prevout_txid = :prevout_txid
                       AND prevout_idx = :prevout_idx
+                      AND addresses.cached_transparent_receiver_address = :address
```

Comment:
I believe this `UPDATE` will silently fail if `:address` is not present in `addresses`?

Reply:
https://www.sqlite.org/lang_update.html
> It is not an error if the WHERE clause does not evaluate to true for any row in the table - this just means that the UPDATE statement affects zero rows.

But yes, a test would be good.

Reply:
Test: https://github.com/zcash/librustzcash/pull/672/commits/306d37e706ae56759ba754b1e35d2d00cb46a04a